libraries/Microsoft.Bot.Connector/Teams/TeamsOperations.cs

```diff
@@ -734,153 +764,142 @@ private async Task<HttpOperationResponse<string>> SendMessageToAllUsersInTenantW
+                tracingParameters.Add("activity", activity);
+                tracingParameters.Add("teamId", teamId);
+                tracingParameters.Add("tenantId", tenantId);
+                tracingParameters.Add("cancellationToken", cancellationToken);
+                ServiceClientTracing.Enter(invocationId, this, "SendMessageToAllUsersInTeam", tracingParameters);
+            }
 
-                    // invalid/unexpected status code
-                    var ex = new HttpOperationException($"Operation returned an invalid status code '{statusCode}'");
-                    if (httpResponse.Content != null)
-                    {
-                        responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        responseContent = string.Empty;
-                    }
+            var content = new
+            {
+                Activity = activity,
+                TeamId = teamId,
+                TenantId = tenantId,
+            };
 
-                    ex.Request = new HttpRequestMessageWrapper(httpRequest, requestContent);
-                    ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
-                    if (shouldTrace)
-                    {
-                        ServiceClientTracing.Error(invocationId, ex);
-                    }
+            var apiUrl = "v3/batch/conversation/team/";
 
-                    throw ex;
-                }
-            }
-            finally
+            return await GetBatchResponseAsync<string>(apiUrl, "POST", shouldTrace, invocationId, activity, content, customHeaders, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<HttpOperationResponse<BatchOperationState>> GetOperationStateWithRetryAsync(string operationId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Tracing
+            var shouldTrace = ServiceClientTracing.IsEnabled;
+            string invocationId = null;
+            if (shouldTrace)
             {
-                if (httpResponse != null)
-                {
-                    httpResponse.Dispose();
-                }
+                invocationId = ServiceClientTracing.NextInvocationId.ToString(CultureInfo.InvariantCulture);
+                var tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("operationId", operationId);
+                tracingParameters.Add("cancellationToken", cancellationToken);
+                ServiceClientTracing.Enter(invocationId, this, "GetOperationState", tracingParameters);
             }
 
+            var apiUrl = "v3/batch/conversation/{operationId}";
+            apiUrl = apiUrl.Replace("{operationId}", Uri.EscapeDataString(operationId));
```

Comment:
For the created urls, could we do something like this? https://siderite.dev/blog/clean-way-to-build-urls-in-net
![image](https://github.com/southworks/botbuilder-dotnet/assets/62260472/943ae702-9bd9-41bb-a9cb-ca357f57a9cb)
It will be reduced in 1 line and reuse for the rest of similar operations like this.

Reply:
No, it reduces the amount of code, but following the way the other methods do it, you can see that they do it for clarity and readability of the code.